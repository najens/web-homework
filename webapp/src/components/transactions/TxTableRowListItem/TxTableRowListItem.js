import React from 'react'
import { bool, string } from 'prop-types'
import { css } from '@emotion/core'
import TxTableCellList from '../TxTableCellList'

export default function TxTableRowListItem ({ categoryName, date, description, dollarAmount, id, indexIsOdd, merchantName, userName }) {
  return (
    <tr css={rowStyle(indexIsOdd)} data-testid={`transaction-${id}`}>
      <TxTableCellList
        categoryName={categoryName}
        date={date}
        description={description}
        dollarAmount={dollarAmount}
        id={id}
        merchantName={merchantName}
        userName={userName}
      />
    </tr>
  )
}

TxTableRowListItem.propTypes = {
  categoryName: string,
  date: string,
  description: string,
  dollarAmount: string,
  id: string,
  indexIsOdd: bool,
  merchantName: string,
  userName: string
}

const rowStyle = (addBackground) => css`
  font-size: 12px;
  line-height: 2;
  ${addBackground && `
  background-color: #e4edf9;
  `}
`
