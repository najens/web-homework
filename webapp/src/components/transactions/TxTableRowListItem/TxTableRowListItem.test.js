import React from 'react'
import { create } from 'react-test-renderer'
import TxTableRowListItem from './TxTableRowListItem'

// Only testing render since this component has no business logic
describe('TxTableRowListItem component', () => {
  it('renders correctly', () => {
    const testRenderer = create(
      <TxTableRowListItem
        categoryName='Restaurant'
        date='Mar 11'
        description='Test'
        dollarAmount='$1.00'
        id='3810c583-1307-4433-8900-8ad6d743f13d'
        indexIsOdd
        merchantName='Taco Bell'
        userName='John Doe'
      />
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
