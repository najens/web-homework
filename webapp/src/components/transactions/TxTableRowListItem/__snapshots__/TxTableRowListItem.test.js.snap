// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxTableRowListItem component renders correctly 1`] = `
.emotion-7 {
  font-size: 12px;
  line-height: 2.5;
  background-color: #e4edf9;
}

.emotion-7:hover {
  cursor: pointer;
}

.emotion-0 {
  border: 1px solid #b4c1ec;
  padding: 0px 6px;
}

.emotion-6 {
  border: 1px solid #b4c1ec;
  padding: 0px 6px;
  direction: rtl;
}

<tr
  className="emotion-7"
  data-testid="transaction-3810c583-1307-4433-8900-8ad6d743f13d"
  onClick={[Function]}
>
  <td
    className="emotion-0"
    data-testid="transaction-3810c583-1307-4433-8900-8ad6d743f13d-date"
  >
    Mar 11
  </td>
  <td
    className="emotion-0"
    data-testid="transaction-3810c583-1307-4433-8900-8ad6d743f13d-userName"
  >
    John Doe
  </td>
  <td
    className="emotion-0"
    data-testid="transaction-3810c583-1307-4433-8900-8ad6d743f13d-description"
  >
    Test
  </td>
  <td
    className="emotion-0"
    data-testid="transaction-3810c583-1307-4433-8900-8ad6d743f13d-merchantName"
  >
    Taco Bell
  </td>
  <td
    className="emotion-0"
    data-testid="transaction-3810c583-1307-4433-8900-8ad6d743f13d-category"
  >
    Restaurant
  </td>
  <td
    className="emotion-0"
    data-testid="transaction-3810c583-1307-4433-8900-8ad6d743f13d-filler"
  >
    
  </td>
  <td
    className="emotion-6"
    data-testid="transaction-3810c583-1307-4433-8900-8ad6d743f13d-amount"
  >
    1 $ +
  </td>
</tr>
`;
